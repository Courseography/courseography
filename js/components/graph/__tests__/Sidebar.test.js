import { fireEvent } from "@testing-library/react"
import TestSidebar from "./TestSidebar"
import TestContainer from "./TestContainer"

describe("Sidebar", () => {
  it("should match shallow snapshot", async () => {
    const sidebar = await TestSidebar.build()
    expect(sidebar).toMatchSnapshot()
  })

  it("Pressing on the sidebar button should open it", async () => {
    const sidebar = await TestSidebar.build()
    expect(sidebar.getByTestId("test-toggle").classList.contains("collapsed")).toBe(
      true
    )
    expect(sidebar.getByTestId("test-toggle").classList.contains("expanded")).toBe(
      false
    )
    fireEvent.click(sidebar.getByTestId("test-sidebar-button"))
    expect(sidebar.getByTestId("test-toggle").classList.contains("collapsed")).toBe(
      false
    )
    expect(sidebar.getByTestId("test-toggle").classList.contains("expanded")).toBe(true)
  })

  it("Clicking a graph button should increase the FCE Count", async () => {
    const container = await TestContainer.build()
    expect(container.getByText("FCE Count: 0.0")).toBeDefined()
    fireEvent.click(container.getByTestId("aaa100"))
    expect(container.getByText("FCE Count: 0.5")).toBeDefined()
    fireEvent.click(container.getByTestId("aaa101"))
    expect(container.getByText("FCE Count: 1.0")).toBeDefined()
    fireEvent.click(container.getByTestId("aaa101"))
    expect(container.getByText("FCE Count: 0.5")).toBeDefined()
  })

  it("Clicking a graph button adds and removes the course to the Selected Courses", async () => {
    const container = await TestContainer.build()
    fireEvent.click(container.getByTestId("aaa100"))
    const sidebar = await TestSidebar.build()
    expect(sidebar.getByTestId("test AAA100")).toBeDefined()
    fireEvent.click(container.getByTestId("aaa100"))
    expect(sidebar.queryByTestId("test AAA100")).toBeNull()
  })

  it("Clicking graph buttons add and remove the courses to the Selected Courses", async () => {
    const container = await TestContainer.build()
    fireEvent.click(container.getByTestId("aaa100"))
    fireEvent.click(container.getByTestId("aaa303"))
    const sidebar = await TestSidebar.build()
    expect(sidebar.getByTestId("test AAA100")).toBeDefined()
    expect(sidebar.getByTestId("test AAA303")).toBeDefined()
    fireEvent.click(container.getByTestId("aaa100"))
    fireEvent.click(container.getByTestId("aaa303"))
    expect(sidebar.queryByTestId("test AAA100")).toBeNull()
    expect(sidebar.queryByTestId("test AAA303")).toBeNull()
  })

  it("Clicking the reset selections clears the Selected Courses and resets FCE", async () => {
    const container = await TestContainer.build()
    fireEvent.click(container.getByTestId("aaa100"))
    fireEvent.click(container.getByTestId("test-reset"))
    expect(container.getByText("FCE Count: 0.0")).toBeDefined()
    const sidebar = await TestSidebar.build()
    expect(sidebar.queryByTestId("test AAA100")).toBeNull()
  })

  it("Searching in the Search bar yields the correct result", async () => {
    const sidebar = await TestSidebar.build()
    const input = sidebar.getByTestId("test-search-bar")
    fireEvent.change(input, { target: { value: "AAA100" } })
    expect(input.value).toBe("AAA100")
  })

  it("Search bar dropdown is only rendered to DOM once an input is recognized", async () => {
    const sidebar = await TestSidebar.build()
    expect(sidebar.queryByTestId("test-searchDropdown")).toBeNull()
    const input = sidebar.getByTestId("test-search-bar")
    fireEvent.change(input, { target: { value: "AAA100" } })
    expect(input.value).toBe("AAA100")
    expect(sidebar.queryByTestId("test-searchDropdown")).toBeDefined()
  })

  // it('Clicking an item from the Search bar updates FCE count and adds it to the active courses', async () => {
  //   const sidebar = await TestSidebar.build();
  //   const input = sidebar.getByTestId('test-search-bar');
  //   fireEvent.change(input, { target: { value: 'AAA100' } });
  //   const container = await TestContainer.build();
  //   expect(container.getByText("FCE Count: 0.0")).toBeDefined();
  //   const liArr = sidebar.getAllByLabelText('test-li');
  //   expect(liArr[0].textContent).toBe('AAA100');
  //   fireEvent.click(liArr[0]);
  //   expect(container.getByText("FCE Count: 0.5")).toBeDefined();
  //   expect(sidebar.getByTestId("test aaa100")).toBeDefined();
  //   fireEvent.click(sidebar.getByTestId("test-search-AAA100"));
  //   expect(sidebar.queryByTestId("test aaa100")).toBeNull();
  //   expect(container.getByText("FCE Count: 0.0")).toBeDefined();
  // });
})
