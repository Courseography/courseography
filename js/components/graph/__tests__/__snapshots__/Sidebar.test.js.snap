// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should match shallow snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    

    <nav>
      
    
      <ul>
        
        
        <li
          id="nav-graph"
        >
          
            
          <a
            href="/graph"
          >
            Graph
          </a>
          
        
        </li>
        
    
      </ul>
      

    </nav>
    

    <div
      class="react-graph"
      id="react-graph"
    />
    

    <div
      id="fcecount"
    />
    <div>
      <div
        class="collapsed sidebar"
        data-testid="test-toggle"
      >
        <div
          class="fcecount"
          data-testid="test-fcecount"
        >
          FCE Count: 
        </div>
        <div
          class="sidebar-dropdown"
          data-testid="test-sidebar"
        >
          <div>
            <label
              for="header-search"
            >
              <span
                class="label-hidden"
              >
                Search courses
              </span>
            </label>
            <input
              class="search-bar"
              data-testid="test-search-bar"
              id="header-search"
              type="text"
            />
          </div>
          <h3
            class="selected-courses"
          >
            Selected courses
          </h3>
          <div
            class="courses"
            data-testid="test-course-selection"
          />
          <button
            class="reset-selections"
            data-testid="test-reset"
          >
            Reset Selections
          </button>
        </div>
        <div
          class="sidebar-button"
          data-testid="test-sidebar-button"
        >
          <img
            id="sidebar-icon"
            src="/static/res/ico/sidebar.png"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="collapsed sidebar"
      data-testid="test-toggle"
    >
      <div
        class="fcecount"
        data-testid="test-fcecount"
      >
        FCE Count: 
      </div>
      <div
        class="sidebar-dropdown"
        data-testid="test-sidebar"
      >
        <div>
          <label
            for="header-search"
          >
            <span
              class="label-hidden"
            >
              Search courses
            </span>
          </label>
          <input
            class="search-bar"
            data-testid="test-search-bar"
            id="header-search"
            type="text"
          />
        </div>
        <h3
          class="selected-courses"
        >
          Selected courses
        </h3>
        <div
          class="courses"
          data-testid="test-course-selection"
        />
        <button
          class="reset-selections"
          data-testid="test-reset"
        >
          Reset Selections
        </button>
      </div>
      <div
        class="sidebar-button"
        data-testid="test-sidebar-button"
      >
        <img
          id="sidebar-icon"
          src="/static/res/ico/sidebar.png"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
